paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/write/BaseV2WriteBuilder.scala

```diff
@@ -32,6 +33,15 @@ abstract class BaseV2WriteBuilder(table: Table)
   protected var overwriteDynamic = false
   protected var overwritePartitions: Option[Map[String, String]] = None
 
+  protected var isOverwriteFiles = false
+  protected var copyOnWriteScan: Option[PaimonCopyOnWriteScan] = None
+
+  def overwriteFiles(scan: Option[PaimonCopyOnWriteScan]): WriteBuilder = {
+    this.isOverwriteFiles = true
```

Comment:
Shall we assert scan is not empty? Or only scan is not empty, then set `isOverwriteFiles` and `copyOnWriteScan`.